newrelic/structures_newrelic_alert_muting_rule.go

```diff
@@ -33,6 +37,17 @@ func expandMutingRuleCreateInput(d *schema.ResourceData) (alerts.MutingRuleCreat
 	return createInput, nil
 }
 
+func expandMutingRuleEndBehaviour(endBehaviour string) alerts.AlertsActionOnMutingRuleWindowEnded {
+	switch endBehaviour {
+	case string(alerts.AlertsActionOnMutingRuleWindowEndedTypes.CLOSE_ISSUES_ON_INACTIVE):
+		return alerts.AlertsActionOnMutingRuleWindowEndedTypes.CLOSE_ISSUES_ON_INACTIVE
+	case string(alerts.AlertsActionOnMutingRuleWindowEndedTypes.DO_NOTHING):
+		return alerts.AlertsActionOnMutingRuleWindowEndedTypes.DO_NOTHING
+	default:
+		return alerts.AlertsActionOnMutingRuleWindowEndedTypes.CLOSE_ISSUES_ON_INACTIVE
+	}
+}
+
```

Comment:
@Aashirwadjain with the change I've mentioned above to the ValidateFunc, this function would no longer be needed, as the `ValidateFunc` would now make sure a "string" is sent, that too, only with the two valid values as imported from the right type from the `alerts` package in the Go Client.

This function can be discarded, and areas calling this function can be converted as shown below.

**from**
```
	if endBehaviour, ok := d.GetOk("end_behaviour"); ok {
		createInput.ActionOnMutingRuleWindowEnded = expandMutingRuleEndBehaviour(endBehaviour.(string))
	}
```

**to**
```
	if endBehaviour, ok := d.GetOk("end_behaviour"); ok {
		createInput.ActionOnMutingRuleWindowEnded = endBehaviour
	}
```
(remember to change the attribute name as specified in the first comment)

Reply:
Removed `expandMutingRuleEndBehaviour` method and implemented typecasting for `alerts.AlertsActionOnMutingRuleWindowEnded`